lib/Tools/circt-lec/ConstructLEC.cpp

```diff
@@ -69,14 +69,9 @@ void ConstructLECPass::runOnOperation() {
   // Create necessary function declarations and globals
   OpBuilder builder = OpBuilder::atBlockEnd(getOperation().getBody());
   Location loc = getOperation()->getLoc();
-  auto ptrTy = LLVM::LLVMPointerType::get(builder.getContext());
-  auto voidTy = LLVM::LLVMVoidType::get(&getContext());
 
-  // Lookup or declare printf function.
-  auto printfFunc = LLVM::lookupOrCreateFn(builder, getOperation(), "printf",
-                                           ptrTy, voidTy, true);
-  if (failed(printfFunc)) {
-    getOperation()->emitError("failed to lookup or create printf");
+  if (insertMainFunc && (!insertReporting)) {
```

Comment:
Maybe `insertMainFunc` and `insertReporting` should be merged into one enum-option with 3 enumeration cases (no insertion, reporting, reporting+main)? That way the user isn't able to specify such an invalid combination in the first place. There are examples of such enum options in `ArcPasses.td` and `FIRRTL/Passes.td`.